monai/apps/nuclick/transforms.py

```diff
@@ -0,0 +1,506 @@
+    12 small regions of 1's it will delineate them into 12 different label classes
+    """
+
+    def __init__(self, keys: KeysCollection):
+        super().__init__(keys)
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            d[key] = measure.label(d[key], connectivity=1).astype(np.uint8)
+        return d
+
+
+class ExtractPatchd(MapTransform):
+    """
+    Extracts a patch from the given image and label, however it is based on the centroid location.
+    The centroid location is a 2D coordinate (H, W). The extracted patch is extracted around the centroid,
+    if the centroid is towards the edge, the centroid will not be the center of the image as the patch will be
+    extracted from the edges onwards
+
+    Args:
+        keys: image, label
+        centroid_key: key where the centroid values are stored, defaults to ``"centroid"``
+        patch_size: size of the extracted patch
+    """
+
+    def __init__(self, keys: KeysCollection, centroid_key: str = NuclickKeys.CENTROID.value, patch_size: int = 128):
+        super().__init__(keys)
+        self.centroid_key = centroid_key
+        self.patch_size = patch_size
+
+    def __call__(self, data):
+        d = dict(data)
+
+        centroid = d[self.centroid_key]  # create mask based on centroid (select nuclei based on centroid)
+        roi_size = (self.patch_size, self.patch_size)
+
+        for key in self.keys:
+            img = d[key]
+            x_start, x_end, y_start, y_end = self.bbox(self.patch_size, centroid, img.shape[-2:])
+            cropped = img[:, x_start:x_end, y_start:y_end]
+            d[key] = SpatialPad(spatial_size=roi_size)(cropped)
+        return d
+
+    def bbox(self, patch_size, centroid, size):
+        x, y = centroid
+        m, n = size
+
+        x_start = int(max(x - patch_size / 2, 0))
+        y_start = int(max(y - patch_size / 2, 0))
+        x_end = x_start + patch_size
+        y_end = y_start + patch_size
+        if x_end > m:
+            x_end = m
+            x_start = m - patch_size
+        if y_end > n:
+            y_end = n
+            y_start = n - patch_size
+        return x_start, x_end, y_start, y_end
+
+
+class SplitLabeld(MapTransform):
+    """
+    Extracts a single label from all the given classes, the single label is defined by mask_value, the remaining
+    labels are kept in others
+
+    Args:
+        label: label source, defaults to ``"label"``
+        others: other labels storage key, defaults to ``"others"``
+        mask_value: the mask_value that will be kept for binarization of the label, defaults to ``"mask_value"``
+        min_area: The smallest allowable object size.
+    """
+
+    def __init__(
+        self,
+        label: str = NuclickKeys.LABEL.value,
+        others: str = NuclickKeys.OTHERS.value,
+        mask_value: str = NuclickKeys.MASK_VALUE.value,
+        min_area: int = 5,
+    ):
```

Comment:
Not sure if it makes sense for this one.

Reply:
Hi @finalelement ,

I think it's better to use `keys` as we inherit from `MapTransform` here. How about:
1. Use `keys` instead of the `label` arg.
2. Add a check in the code that `len(keys) == 1`.
3. No need to support `allow_missing_keys`, just set to `False` for super().

@drbeh What do you think?

Thanks.

Reply:
Added as per suggestion